--- a/src/ap/ap_config.h
+++ b/src/ap/ap_config.h
@@ -1162,6 +1162,11 @@
 	u8  ht_cap_ampdu_parameters_override_mask;
 	u32 ht_mcs_set[4];
 
+
+	int noscan;
+	int no_ht_coex;
+
+
 	int ieee80211n;
 	int ieee80211n_acax_compat;
 	int secondary_channel;
--- a/hostapd/config_file.c
+++ b/hostapd/config_file.c
@@ -4853,6 +4853,16 @@
 		if (bss->ocv && !bss->ieee80211w)
 			bss->ieee80211w = 1;
 #endif /* CONFIG_OCV */
+
+
+	} else if (os_strcmp(buf, "noscan") == 0) {
+		conf->noscan = atoi(pos);
+	} else if (os_strcmp(buf, "ht_coex") == 0) {
+		conf->no_ht_coex = !atoi(pos);
+
+
+
+
 	} else if (os_strcmp(buf, "ieee80211n") == 0) {
 		int val = atoi(pos);
 		if (!IS_VALID_BOOL_RANGE(val)) {
--- a/src/ap/dfs.c	2021-07-06 04:54:34.000000000 +0800
+++ b/src/ap/dfs.c	2021-08-01 13:53:14.474028871 +0800
@@ -1073,6 +1073,8 @@
 		return 0;
 	}
 
+	iface->dfs_cac_ms=1000;
+
 	/* Finally start CAC */
 	hostapd_set_state(iface, HAPD_IFACE_DFS);
 	wpa_printf(MSG_DEBUG, "DFS start CAC on %d MHz", iface->freq);
--- a/src/ap/hw_features.c
+++ b/src/ap/hw_features.c
@@ -724,7 +724,8 @@
 	int ret;
 
 	/* Check that HT40 is used and PRI / SEC switch is allowed */
-	if (!iface->conf->secondary_channel || iface->conf->no_pri_sec_switch)
+	if (!iface->conf->secondary_channel || iface->conf->no_pri_sec_switch ||
+		iface->conf->noscan)
 		return 0;
 
 #ifdef CONFIG_ACS
--- a/src/ap/ieee802_11_ht.c
+++ b/src/ap/ieee802_11_ht.c
@@ -362,6 +362,11 @@
 			   "Ignore 20/40 BSS Coexistence Management frame since 40 MHz capability is not enabled");
 		return;
 	}
+	
+	
+	if (iface->conf->noscan || iface->conf->no_ht_coex)
+		return;
+
 
 	if (len < IEEE80211_HDRLEN + 2 + sizeof(*bc_ie)) {
 		wpa_printf(MSG_DEBUG,
@@ -691,6 +696,9 @@
 	if (iface->current_mode->mode != HOSTAPD_MODE_IEEE80211G)
 		return;
 
+	if (iface->conf->noscan || iface->conf->no_ht_coex)
+		return;
+
 	if (iface->conf->ignore_40_mhz_intolerant) {
 		wpa_printf(MSG_INFO, "HT: Ignore Forty MHz Intolerant that is set by STA " MACSTR
 				   " in Association Request", MAC2STR(sta->addr));
